src/main/java/com/pinterest/secor/monitoring/MicroMeterMetricCollector.java

```diff
@@ -19,24 +19,28 @@
 package com.pinterest.secor.monitoring;
 
 import com.pinterest.secor.common.SecorConfig;
-
-import com.google.common.util.concurrent.AtomicDouble;
 import io.micrometer.core.instrument.Clock;
 import io.micrometer.core.instrument.MeterRegistry;
 import io.micrometer.core.instrument.Metrics;
 import io.micrometer.core.instrument.Tag;
 import io.micrometer.jmx.JmxConfig;
 import io.micrometer.jmx.JmxMeterRegistry;
+import io.micrometer.prometheus.PrometheusConfig;
+import io.micrometer.prometheus.PrometheusMeterRegistry;
 import io.micrometer.statsd.StatsdConfig;
 import io.micrometer.statsd.StatsdMeterRegistry;
 
 import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
 
 /**
  * MicorMeter meters can integrate with many different metrics backend 
  * (StatsD/Promethus/Graphite/JMX etc, see https://micrometer.io/docs)
  */
 public class MicroMeterMetricCollector implements MetricCollector {
+    private static final Map<String, Double> GAUGE_CACHE = new HashMap<>();
```

Comment:
How big will this cache become?

Reply:
It depends, on how many different gauges will be tracked. Currently it will be 2*topic-count. Maybe I should limit cache size to avoid memory leak, if someone adds way too many metrics with different tags?
The cache is used, because prometheus is returning NaNs otherwise. https://stackoverflow.com/questions/50821924/micrometer-prometheus-gauge-displays-nan

Reply:
I suggest to add a config param for the cache size (e.g. 100), if the cache goes over that size, throw a runtime exception with error message to tell user to increase the cache size parameter.

Reply:
Maybe just log warning message? We probably shouldn't crash writing process because of metric cache overflow.